### ISSUE TITLE
No issue - QR-Code with logo possible?

### ISSUE THREAD

**Issue by Dunkelschunkel** [ufozone/ha-unifi-voucher#72]
I've just today stumbled over your integration and it perfectly fits into my use case at home. I am able to request vouchers via signal message and get one back together with the QR-Code to connect to the WiFi. All together it was a good bit of effort but it is working. It even detects if there are no more vouchers and generatss some new ones first. All in all I am pleased and even more thankful towards you, the mastermind and genius behind this integration. 

Thus said, improving the QR-Code entity with an optional logo or image would be small but tremendous improvement. 

Cheers from Germany 

Dunkelschunkel

**Comment by ufozone** [comment]
Thank you very much for your feedback. 

The logo in the QR code is a great idea. The QR code in my integration was actually just a quick win without much effort, as I use the QR code implementation of the Unifi AIO library:
https://github.com/Kane610/aiounifi/blob/469aabdc478009a48352d8259d2f463598836bff/aiounifi/models/wlan.py#L92

The library in turn uses the Segno library with a ready-made method:
https://segno.readthedocs.io/en/latest/special-qrcode-factories.html

Unfortunately, there is no option to integrate own picture. But it looks to me as if we can do all sorts of things with QR codes using the Sengo Library. So we have to completely rebuild the Wifi QR code thing with the Sengo methods. That looks like a lot of work.

You are welcome to give it a try.

**Comment by Dunkelschunkel** [comment]
@ufozone No coding skills required I hope. Thanks for shedding some light on that topic though. 

Dunkelschunkel

**Comment by ufozone** [comment]
I wanted to try it out now. That's where I am right now:
![grafik](https://github.com/ufozone/ha-unifi-voucher/assets/29913712/05481449-870d-44a6-b388-fe1450d0035b)

I am now making the source code look good again and adding a new setting in setup and options flow.

**Comment by ufozone** [comment]
@Dunkelschunkel , done! But a lot of work and a lot of new dependencies. I can't promise that the feature will remain as it is in the future.

**Comment by ufozone** [comment]
@Dunkelschunkel , done! The feature is available in the main branch and will be released in version 1.2.3 in few weeks. But a lot of work and a lot of new dependencies. I can't promise that the feature will remain as it is in the future.

**Comment by Dunkelschunkel** [comment]
@ufozone ,
Wow, this looks quite pleasant. And all this for a "won't fix". I don't know what to say except I am eager to see and test it by myself. In the meantime I used another QR code integration available via hacs to bring some color into the mix. But I'll switch to your QR code as soon as it goes live. 
Thanks.

**Comment by ufozone** [comment]
@Dunkelschunkel , you can also select the main branch as the source for the integration in HACS. Then you can test the new feature immediately. The new version will be at least two weeks away.

Can you please name the QR code integration you are currently using?
What does your configuration looks like? 
Please include a screenshot of your implementation.

As written, I have introduced a lot of new dependencies with the feature and cannot promise that the feature will remain available in future versions.

I would therefore like to document an alternative solution right away.

**Comment by Dunkelschunkel** [comment]
Hi @ufozone, 

The QR-Code integration is called QR code generator and the link is https://github.com/DeerMaximum/QR-Code-Generator
They can use templates when creating QR-Codes and even create WiFi codes. And some more options are colors, border and size. 
Mine is just a hint of blue I am also using with my hotspot portal. 
![Screenshot_2024-05-27-10-10-14-668_io homeassistant companion android](https://github.com/ufozone/ha-unifi-voucher/assets/83595637/a8debe7e-1019-4b9e-be1e-92708a3953c4)
![Screenshot_2024-05-27-10-10-21-547_io homeassistant companion android](https://github.com/ufozone/ha-unifi-voucher/assets/83595637/af4907c3-e491-4dd3-a1cc-c460c0b88ffc)
![Screenshot_2024-05-27-10-10-34-643_io homeassistant companion android](https://github.com/ufozone/ha-unifi-voucher/assets/83595637/6e03ecd3-9df8-4e3c-8efa-1e281d166add)

**Comment by ufozone** [comment]
Of course, I hadn't thought about the fact that the new function should still be documented. I'll add it to the ToDo list.

The logo just needs to be in a place that Home Assistant can access. It makes no sense to specify a fixed location, as there are too many different installation constellations. The folder `/config/custom_components/unifi_voucher/` is over written when the integration is updated. I recommend the persistent folder `/config/www/`.
